### PR TITLE
filter out incomplete profiles

### LINKED ISSUE #3978: [map] create a filter to return only users with a complete profile
When I search for a destination it is usually full of profiles that are empty, I as a user would like to be able to see only the profiles that have a photo/description.

I would like to make a toggle in the filter modal that makes the backend return only the profiles that have a complete profile.

### Where
image

### Reference
image

### Task
Make the backend to return only complete profiles when the parameter `only_complete_profiles` was sent by the frontned

### PR DESCRIPTION
**Pull request by Krishnag09**
closes #3978 - create a filter to return only users with a complete profile

- [x] Formatted my code by running `autoflake -r -i --remove-all-unused-imports src && isort . && black .` in `app/backend`
- [x] Added tests for any new code or added a regression test if fixing a bug
- [x] All tests pass
- [x] Run the backend locally and it works
- [x] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

### CODE REVIEW

File: app/backend/src/tests/test_search.py [hunk 1]
```diff
@@ -73,3 +77,48 @@ def test_regression_search_in_area(db):
...
+        )
+        session.commit()
+
+    user_complete_profile, token6 = generate_user(about_me="this profile is complete", avatar_key=key)
+
+    user_incomplete_profile, token7 = generate_user(about_me="", avatar_key=key2)
+
+    with search_session(token6) as api:
+        req = search_pb2.UserSearchReq()
+        req.profile_completed = True
```
Comment (aapeliv) [inline app/backend/src/tests/test_search.py]: I think you need to do something like this: `wrappers_pb2.BoolValue(value=True)`, instead of just `True`

Comment (Krishnag09) [reply app/backend/src/tests/test_search.py]: @aapeliv -its finally ready for review. i had to change the assert to accommodate the extra search user result from the uploader user.

File: app/backend/src/tests/test_search.py [hunk 2]
```diff
@@ -51,11 +55,11 @@ def test_regression_search_in_area(db):
...
-    # outside
-    user1, token1 = generate_user(geom=create_coordinate(1, 0), geom_radius=100)
-    # outside
-    user2, token2 = generate_user(geom=create_coordinate(0, 1), geom_radius=100)
-    # inside
+    # # outside
+    # user1, token1 = generate_user(geom=create_coordinate(1, 0), geom_radius=100)
+    # # outside
+    # user2, token2 = generate_user(geom=create_coordinate(0, 1), geom_radius=100)
+    # # inside
```
Comment (aapeliv) [inline app/backend/src/tests/test_search.py]: Why are these commented out?

File: app/backend/src/tests/test_search.py [hunk 3]
```diff
@@ -73,3 +77,50 @@ def test_regression_search_in_area(db):
...
+        key = random_hex(32)
+        filename = random_hex(32) + ".jpg"
+        session.add(
+            Upload(
+                key=key,
+                filename=filename,
+                creator_user_id=uploader_user.id,
+            )
+        )
+        session.commit()
```
Comment (aapeliv) [inline app/backend/src/tests/test_search.py]: This commit is not needed when exiting the context manager, it'll commit automatically

**Review by aapeliv — DISMISSED** [review]
Looks good! Thank you!

**Comment by aapeliv** [comment]
@Krishnag09 did you have a chance to look at the comments?

**Comment by Krishnag09** [comment]
> @Krishnag09 did you have a chance to look at the comments?

yes just pushed changes.

**Comment by aapeliv** [comment]
Great, will have a look

**Review by aapeliv — APPROVED** [review]
Looks great to me! Thanks a lot!

**Comment by aapeliv** [comment]
Also please squash these when you merge!

**Comment by aapeliv** [comment]
It’s live now on prod!

**Pull request merged**